stat_summary_plots.py

```diff
@@ -0,0 +1,116 @@
+import sklearn as sk
+from sklearn.linear_model import LinearRegression
+import numpy as np
+from sklearn import datasets
+from numpy.linalg import det
+from numpy.linalg import inv
+from numpy import mat
+from numpy import random
+import matplotlib.pyplot as plt
+import seaborn as sns
+import pandas as pd
+import scipy.stats as ss
+
+
+class lr:
+    '''
+    implements the linear regression algorithm class
+    '''
+
+    def __init__(self, data_source, separator):
+        self.beta = np.matrix(np.zeros(2))
+        self.yhat = np.matrix(np.zeros(2))
+        self.r2 = 0.0
+        self.se = 0.0
+        self.f = 0.0
+        self.msr = 0.0
+        self.mse = 0.0
+        self.p = 0.0
+        data_mat = np.genfromtxt(data_source, delimiter=separator)
+        self.xarr = data_mat[:, :-1]
+        self.yarr = data_mat[:, -1]
+        self.ybar = np.mean(self.yarr)
+        self.dfd = len(self.yarr) - 2  #df
+        return
+
+
+    def cov_custom(x, y):
+        result = sum((x - np.mean(x)) * (y - np.mean(y))) / (len(x) - 1)
+        return result
+
+
+    def corr_custom(x, y):
+        return lr.cov_custom(x, y) / (np.std(x, ddof=1) * np.std(y, ddof=1))
+
+
+    def simple_regression(self):
+        xmat = np.mat(self.xarr)
+        ymat = np.mat(self.yarr).T
```

Comment:
[Here](https://docs.scipy.org/doc/numpy/reference/generated/numpy.matrix.html#numpy.matrix) is a note from numpy documentation for matrix class: "It is no longer recommended to use this class, even for linear algebra. Instead use regular arrays. The class may be removed in the future." 
You can use an ndarray with transpose method:
```python
self.yarr = np.array(self.yarr)
ymat = self.yarr.transpose()
```